docs/wfe.rst

```diff
@@ -0,0 +1,216 @@
+Representing sources of wavefront error
+=======================================
+
+When used for Fraunhofer-domain propagation, POPPY supports expressing aberrations as if they were applied in a pupil plane. Currently, there is the `~poppy.ZernikeWFE` optical element to represent wavefront error as a combination of Zernike terms, and the `~poppy.ParameterizedWFE` optical element that offers the ability to specify basis sets other than Zernikes.
```

Comment:
Pupil plane is fine as a default, but the point is the math should work just as well for any plane which has dimensions in meters available (e.g. anything other than a Fraunhofer-domain image plane calculation). 

Something to consider for eventual API larger overhaul: In retrospect I don't think that I should have made `planetype` a property of an optic itself. Rather it is a property of the ordering of planes that makes up an optical system.   The `OpticalSystem` should maintain the list of plane types along with the list of plane objects themselves. (And the list of distances between planes, in the case of the Fresnel.)